.github/workflows/build-snap.yml

```diff
@@ -0,0 +1,39 @@
+name: Snap Builder
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main, IENG-* ]
+  # Allow manual trigger
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: 
+          - arm64
+    steps:
+        
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setub QEMU
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: ${{ matrix.platform }}
+
+      - name: Build snap for arm64
+        uses: diddlesnaps/snapcraft-multiarch-action@v1
+        id: build
+        with:
+          architecture: ${{ matrix.platform }}
+          use-podman: true
```

Comment:
> Note that when using Podman your build cannot target a CPU architecture that is different to the host your runner is based upon. This is a limitation of Podman. For example, you will need an ARM64 host to build an ARM64 Snap Package, when using Podman builds.
https://github.com/diddlesnaps/snapcraft-multiarch-action?tab=readme-ov-file#use-podman

Podman will not work for compiling code on amd64 for arm64.